### ISSUE TITLE
Graceful termination is not working for ScyllaCluster containers

### ISSUE THREAD

**Issue by tnozicka** [scylladb/scylla-operator#2239]
I was tracing a case where a startup probe fails and the container gets terminated (empty cluster) and the graceful termination took almost 15m.

```
Dec 04 17:14:48 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:14:48.521667   38896 prober.go:107] "Probe failed" probeType="Startup" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" probeResult="failure" output="HTTP probe failed with statuscode: 503"
Dec 04 17:14:58 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:14:58.522218   38896 prober.go:107] "Probe failed" probeType="Startup" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" probeResult="failure" output="HTTP probe failed with statuscode: 503"
Dec 04 17:14:58 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:14:58.523233   38896 kuberuntime_manager.go:1038] "Message for Container of pod" containerName="scylla" containerStatusID={"Type":"containerd","ID":"c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb"} pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" containerMessage="Container scylla failed startup probe, will be restarted"
Dec 04 17:14:58 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:14:58.523776   38896 kuberuntime_manager.go:1100] "Killing unwanted container for pod" containerName="scylla" containerID={"Type":"containerd","ID":"c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb"} pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0"
Dec 04 17:14:58 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:14:58.523804   38896 kuberuntime_container.go:627] "Running preStop hook" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" containerID="containerd://c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb"
Dec 04 17:15:13 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]:  > execCommand=["/usr/bin/bash","-euExo","pipefail","-O","inherit_errexit","-c","trap 'rm /mnt/shared/ignition.done' EXIT\nnodetool drain \u0026\nsleep 15 \u0026\nwait"] containerName="scylla" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" message=<
Dec 04 17:15:13 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]:  > pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" containerID="containerd://c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb"
Dec 04 17:15:13 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:15:13.553585   38896 kuberuntime_container.go:647] "PreStop hook completed" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" containerID="containerd://c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb"
Dec 04 17:15:13 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:15:13.553605   38896 kuberuntime_container.go:745] "Killing container with a grace period" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" containerID="containerd://c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb" gracePeriod=885
Dec 04 17:29:58 gke-so-753fb68c-ed81-4a23-aa2-workers-24a06bcc-636c kubelet[38896]: I1204 17:29:58.725803   38896 kuberuntime_container.go:754] "Container exited normally" pod="e2e-test-nodeconfig-sf5hl-0-ss4tc/basic-rk6zh-us-east-1-us-east-1a-0" podUID="b01f1990-37ff-4da0-8a50-1b39c3024663" containerName="scylla" containerID="containerd://c3d9dd8656b21c701f5ead2387f30d898a1ebc6a58b5d9edf982f184b28033eb"
```

**Comment by rzetelskik** [comment]
@tnozicka I've hit this a while ago when debugging containers for pausing clusters. Signals were being eaten by the sleep command in a loop and not propagated. I fixed it by adding a SIGINT and SIGTERM trap.

You'll also hit the same issue with scylla-manager-agent container (and a missing exec).

Feel free to test against https://github.com/rzetelskik/scylla-operator/commit/8cc264e9a0f5ba7bb3930480bddb66cae8193827. I can send a PR unless you have a different approach in mind.

**Comment by tnozicka** [comment]
I was thinking to do sleep / wait as we have done before
https://github.com/scylladb/scylla-operator/blob/2ea074e87800c5101111882a687b2826e26f96e5/pkg/controller/scylladbdatacenter/resource.go#L844-L845

**Comment by tnozicka** [comment]
Fixes in #2212
